OMGVA PoS/Business layer/Controllers/BusinessController.cs

```diff
@@ -0,0 +1,86 @@
+{
+    [Route("business")]
+    [ApiController]
+    public class BusinessController(IBusinessRepository businessRepository) : Controller
+    {
+        private readonly IBusinessRepository _businessRepository = businessRepository;
+
+        [HttpGet]
+        [Authorize(Roles = "Admin")]
+        [ProducesResponseType<List<Business>>(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        public IActionResult GetAllBusinesses() {
+            return Ok(JsonConvert.SerializeObject(_businessRepository.GetBusinesses()));
+        }
+
+
+        [HttpGet("{id}")]
+        [Authorize(Roles = "Admin,Owner")]
+        [ProducesResponseType<Business>(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public IActionResult GetBusiness(long id) {
+            JwtSecurityToken token = JwtTokenHelper.GetJwtToken(HttpContext.Request.Headers.Authorization);
+            if (token.UserRoleEquals(UserRole.Owner) && !token.UserBusinessEquals(id)) {
+                return Forbid();
+            }
+            Business business = _businessRepository.GetBusiness(id);
+
+            if (business == null)
+                return NotFound();
+            else
+                return Ok(JsonConvert.SerializeObject(business));
+        }
+
+        [HttpPost]
+        [Authorize(Roles = "Admin")]
+        [ProducesResponseType<Business>(StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        public IActionResult CreateBusiness([FromBody]CreateBusinessRequest createBusinessRequest) {
+            if (!createBusinessRequest.Email.IsValidEmail() || !createBusinessRequest.Phone.IsValidPhone()){
+                return BadRequest();
+            }
+            Business business = _businessRepository.CreateBusiness(createBusinessRequest);
+            return Created($"/business/{business.Id}", business);
+        }
+
+        [HttpPatch("{id}")]
+        [Authorize(Roles = "Admin,Owner")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public IActionResult UpdateBusiness([FromBody] Business business, long id) {
+            JwtSecurityToken token = JwtTokenHelper.GetJwtToken(HttpContext.Request.Headers.Authorization);
+            if (token.UserRoleEquals(UserRole.Owner) && !token.UserBusinessEquals(id)) {
```

Comment:
Do we need to validate user role, if we already specify the roles in an attribute? Why are authorized roles "Admin,Owner" but here we check only for owner? 

```.cs
        [Authorize(Roles = "Admin,Owner")]
```